lib/ps_map.fz

```diff
@@ -316,14 +316,14 @@ O(n log n).
   # both ps_map.this and other, it will be mapped to ps_map.this[k] or to other[k],
   # but it is undefined to which of these two.
   #
-  union (other ref ps_map PK V) ref ps_map PK V is  # NYI: #167: 'ref' due to lack of 'like this'
+  union (other ps_map PK V) ps_map PK V is
     if other.size < size
-      other ∪ ps_map.this
+      other ∪ (ps_map PK V data size fill)
```

Comment:
Ideally, `ps_map.this` should be usable when the argument type if `ps_map.this.type`.